sklearn/feature_selection/tests/test_rfe.py

```diff
@@ -108,6 +108,43 @@ def test_rfe():
     assert_array_almost_equal(X_r, X_r_sparse.toarray())
 
 
+def test_rfe_sample_weights():
+    iris = load_iris()
+    X = iris.data
+    y = iris.target
+
+    clf = SVC(kernel="linear")
+    rfe = RFE(estimator=clf, n_features_to_select=1)
+
+    sample_weight_test = 2
+    class_test = 2
+
+    # Case 1 - original dataset
+    rfe.fit(X, y)
+    ranking_original = rfe.ranking_.copy()
+
+    # Case 2 - double the weight of one class's samples
+    w = np.ones(y.shape[0])
+    w[y == class_test] = sample_weight_test
+
+    rfe.fit(X, y, sample_weight=w)
+    ranking_weights = rfe.ranking_.copy()
+
+    # Case 3 - duplicate the samples of one class
+    extra_X = np.tile(X[y == class_test], (sample_weight_test - 1, 1))
```

Comment:
I am not sure that we need to call `tile` here. Getting a subset of `X` containing the targeted class data could be directly concatenated in `X_duplicate`.

Reply:
Honestly, I'm not sure what the original author meant here... Testing locally, this `tile` call was a no-op, that is

```python
X[y == class_test] == np.tile(X[y == class_test], (sample_weight_test - 1, 1))
```

Reply:
ended up doing what you suggested:

```python
X_duplicate = np.concatenate((X, X[y == class_targeted]), axis=0)
```